src/methods/join.js

```diff
@@ -914,4 +914,49 @@ export class JoinImplementation {
       return d3;
     }
   }
+
+  mapAndJoin(
+    realm: Realm,
+    values: Set<ConcreteValue>,
+    c: ConcreteValue => Value,
```

Comment:
`c` and `f` --- impossible to figure out what's going on unless one studies the code...

How about `terminationConditionCreator` and `indexedMutator`?